### ISSUE TITLE
Exposure curve uses arbitrary 20x EV upper bound instead of distribution quantile

### ISSUE THREAD

**Issue by AlexFiliakov** [AlexFiliakov/Ergodic-Insurance-Limits#1144]
## Problem Statement

`LayerPricer.exposure_curve()` uses `max_loss = ev * 20.0` as the maximum possible loss for the exposure curve calculation. For heavy-tailed distributions (Pareto with alpha near 2, or GPD with positive shape), 20x the expected value may exclude a significant probability mass in the tail, producing an exposure curve that systematically underestimates the proportion of losses retained at high attachment points. For lighter-tailed distributions (lognormal with low CV), 20x EV may be unnecessarily large, wasting resolution in the upper portion of the curve.

## v1.0 Impact

**Medium** — Exposure curves (Lee diagrams) are used for visualizing retention decisions and for exposure rating. An inaccurate upper bound distorts the curve shape, which can mislead users about the risk retained vs. ceded at different attachment points.

## Affected Code

- `ergodic_insurance/insurance_pricing.py`, lines 305-307 (`exposure_curve` method)

```python
# Use a practical upper bound: 20x expected value
max_loss = ev * 20.0
lev_max = self.severity.limited_expected_value(max_loss)
```

## Current Behavior

The exposure curve is normalized against `LEV(20 * EV)`. For a Pareto(2.5, 100K) distribution with EV = 166,667, the upper bound is ~3.33M. The actual 99.9th percentile is ~31.6M, meaning the curve omits tail losses above 3.33M. For a lognormal(mu=10, sigma=1) with EV ~22K, the upper bound is ~440K while the 99.9th percentile is ~485K — more reasonable but still arbitrary.

## Expected Behavior

The upper bound should be determined by the distribution itself, using either:

1. **A high quantile** of the distribution (e.g., 99.9th or 99.99th percentile): `max_loss = distribution.quantile(0.999)`, or
2. **The analytical E[X]** when finite (for the full, uncapped distribution), normalizing the exposure curve as `G(r) = LEV(r * M) / E[X]`, which avoids the need for a finite M altogether, or
3. **A user-configurable parameter** with a sensible default derived from the distribution shape.

Each `LossDistribution` subclass could implement a `quantile()` method (all three current distributions — lognormal, Pareto, GPD — have analytical quantile functions).

## Actuarial References

- Lee (1988), *Loss Distributions* — The exposure curve is defined as `G(r) = LEV(r*M) / LEV(M)` where M is the maximum possible loss or E[X] for unbounded distributions.
- Bernegger (1997), "The Swiss Re Exposure Curves and the MBBEFD Distribution Class" — Proper exposure curves are normalized against the theoretical expected value, not an arbitrary multiple.
- CAS Monograph No. 1, *Increased Limits Ratemaking* — Exposure curves must accurately represent the entire loss distribution to support sound ratemaking.
